x-pack/legacy/plugins/rollup/server/lib/call_with_request_factory/call_with_request_factory.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ElasticsearchServiceSetup } from 'kibana/server';
+import { once } from 'lodash';
+import { elasticsearchJsPlugin } from '../../client/elasticsearch_rollup';
+
+const callWithRequest = once((elasticsearchService: ElasticsearchServiceSetup) => {
+  const config = { plugins: [elasticsearchJsPlugin] };
+  return elasticsearchService.createClient('rollup', config);
+});
+
+export const callWithRequestFactory = (
```

Comment:
Do we need this factory? In my PR, I've used it directly from the router context

https://github.com/elastic/kibana/pull/56829/files#diff-178c0726cd4b1e7299f257e5069b4a42R28

Reply:
Thanks for sharing! I'm going to leave as is for now, but will consider changing this in my next PR.